doc/Overview/Validation/ValidationOverview.md

```diff
@@ -1,16 +1,158 @@
+
+### Code example
+
+An example using validation attributes is shown below:
 
-## Reference
+```csharp
+public class Person : ObservableValidator
+{
+    [Required]
+    public string FirstName { get; set; }
 
-- [For more information on FluentValidation](https://fluentvalidation.net/).
+    [Required]
+    public string LastName { get; set; }
+
+    [Range(0, 100)]
+    public int Age { get; set; }
+}
+```
+
+It is also possible to define custom validation attributes by deriving from the `ValidationAttribute` class. More information on custom validation can be found [here](https://nicksnettravels.builttoroam.com/custom-validation/).
+
+For more information, see the reference documentation about [INotifyDataErrorInfo](https://learn.microsoft.com/dotnet/api/system.componentmodel.inotifydataerrorinfo) and [ObservableValidator](https://docs.microsoft.com/windows/communitytoolkit/mvvm/observablevalidator).
+
+# [**IValidatableObject**](#tab/validatable-object)
+Another option that was developed more recently is the `IValidatableObject` interface. Like `INotifyDataErrorInfo`, it is implemented by a class to ensure properties can be validated based on the attributes they are decorated with. Validator support for this interface is provided internally with the `TryValidateObject` method used by the default validator to execute validation rules on the entity in question. This method returns an `IEnumerable<ValidationResult>` which contains elements which indicate whether the object is valid or not. 
+
+Unlike other options, the validation behavior for `IValidatableObject` based types comes from the `Validate` method implementation where it is typical to invoke the `TryValidateProperty` method. The default validator is responsible for invoking this `Validate` method.
+
+### Code example
+
+```csharp
+public class Person : IValidatableObject
+{
+    [Required]
+    public string FirstName { get; set; }
+
+    [Required]
+    public string LastName { get; set; }
+
+    [Range(0, 100)]
+    public int Age { get; set; }
+
+    public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
+    {
+        var results = new List<ValidationResult>();
+
+        if (!TryValidateProperty(FirstName, new ValidationContext(this, null, null) { MemberName = nameof(FirstName) }, results))
+        {
+            results.Add(new ValidationResult("First name is invalid."));
+        }
+        
+        if (!TryValidateProperty(LastName, new ValidationContext(this, null, null) { MemberName = nameof(LastName) }, results))
+        {
+            results.Add(new ValidationResult("Last name is invalid."));
+        }
```

Comment:
The `IValidatableObject` is not there to replace the attributes, but rather to complement them with fine-grained validation when the out-the-box ones don't cater for.
Consier using a more specific validation error in your example, such as 'name doesn't contain unicode characters' or whatever the validation attributes doesn't cover.

There's also the [`CustomValidationAttribute`](https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.dataannotations.customvalidationattribute) which you might consider mentioning, although not critical.

Reply:
@weitzhandler This is really good feedback! I think I should also give that objective a mention in text alongside changes to the code sample.

Reply:
Not sure if you have to spend too much on it, as this is DataAnnotations specific, just make sure you explain how to harness Uno Extensions' power to it.